include/sys/arc.h

```diff
@@ -157,8 +157,11 @@ typedef enum arc_flags
 	ARC_FLAG_COMPRESS_3		= 1 << 27,
 	ARC_FLAG_COMPRESS_4		= 1 << 28,
 	ARC_FLAG_COMPRESS_5		= 1 << 29,
-	ARC_FLAG_COMPRESS_6		= 1 << 30
-
+	ARC_FLAG_COMPRESS_6		= 1 << 30,
+	/*
+	 * cache lookup only
+	 */
+	ARC_FLAG_CACHED_ONLY	= 1 << 31
```

Comment:
IMHO, this is another commit that should just be merged into #9954, because the only reason cached-only lookups were added is so that the dbuf state machine could determine whether to immediately issue an asynchronous read to resolve the assumed future state of the block for the inbound write.  And as an optimization, it only does this for incomplete writes.